### PR TITLE
make distributed true for multiple process

### PR DESCRIPTION
**Pull request by allanj**
This PR is to fix the issue in https://github.com/huggingface/trl/issues/960#issuecomment-1810865206

The issue will cause "gather stat" not to work for other distributed scenarios other than `MULTI_GPU`.

### CODE REVIEW

File: trl/trainer/ppo_trainer.py [hunk 1]
```diff
@@ -330,7 +330,7 @@ def __init__(
         # In a distributed setup, only logging needs to be performed on the main process
         # check: https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html
         # or: https://discuss.pytorch.org/t/use-distributed-data-parallel-correctly/82500/11
-        self.is_distributed = self.accelerator.distributed_type == "MULTI_GPU"
+        self.is_distributed = self.accelerator.num_processes > 0
```
Comment (younesbelkada) [inline trl/trainer/ppo_trainer.py:333]:
```suggestion
        self.is_distributed = self.accelerator.num_processes > 1
```

**Review by younesbelkada — COMMENTED** [review]
Thanks for the fix! I left one comment, what do you think?

File: trl/trainer/ppo_trainer.py (hunk 1 above)
Comment (allanj) [reply trl/trainer/ppo_trainer.py:333]: Oh, yes. That's correct. My bad

**Comment by allanj** [comment]
It's kind of my first time to make PR here. So I guess one more review needed? @younesbelkada

**Comment by younesbelkada** [comment]
No this is great! Merging!

**Review by younesbelkada — APPROVED** [review]

**Pull request merged**